### DISCUSSION TITLE
Impediments to releasing the standard

### DISCUSSION THREAD

**Discussion by aucampia** [json-schema-org/community#25] [Vision and direction]
As far as I can see, the only defined milestones are for the next drafts. Why were the previous drafts not finalized, and what exactly is needed to get to a release of the standard?

**Comment by Relequestual** [comment]
I'm actually doing a talk about this on Thursday THIS WEEK at apidays Interface. I highly recomend you catch it.
I'm still finishing it, but it covers the story of how we got to where we are now, and what we need to do next.

You may find it interesting to know the current maintainers have only been the maintainers for the last 5 years, draft-05 onwards.

**Reply by michaelmior** [reply]
@Relequestual Will the talk be publicly available for later viewing? I'd be interested in seeing it, but probably can't catch it live.

**Reply by Relequestual** [reply]
I'm pretty sure it will be!

The conference is doing 3 streams in three time zones, but that doesn't help if the day is busy.

**Reply by aucampia** [reply]
Thanks for the good talk, I think maybe the best way to resolve the situation is to clarify ` The Path to Standardization` on the site, I will make a PR for this when I have time (hopefully over the weekend) with my understanding of the current situation, and then see what others think, I am sure it will take some time to finalize the position but it may be worth it, at least to not get misdirected questions all the time.

**Reply by Relequestual** [reply]
Thanks, and thanks for your willingness to contribute. I look forward to reviewing your PR.
If I don't review it in a timley way, please nag me.

**Comment by jdesrosiers** [comment]
I think this is a misunderstanding of how JSON Schema uses the term "draft". What we call a "draft" _is_ a release and is not the same thing as an IETF draft. Sometimes we have more than one IETF draft per release draft. I know it's confusing. We only use that nomenclature for historical reasons and we probably should move away from that to avoid this kind of misunderstanding. JSON Schema will always be an evolving thing like anything else in software. It will never be "done".

**Reply by aucampia** [reply]
> JSON Schema will always be an evolving thing like anything else in software. It will never be "done".

I understand this, I am not really that concerned or interested with JSON schema become an IETF internet standard, I don't even think it is particularly appropriate for something like JSON schema, but json-schema.org expresses this as the goal: 

https://json-schema.org/#the-path-to-standardization

> The JSON Schema project intends to shepherd all three draft series to either: RFC status, the equivalent within another standards body, and/or join a foundation and establish self publication rules.

The site continues:

> The intention, particularly for vocabularies such as validation which have been widely implemented, is to remain as compatible as possible from draft to draft. However, these are still drafts, and given a clear enough need validated with the user community, major changes can occur.

Most of the world is already using JSON schema as if it is not a draft, and everyone should realizes and accept that standards do evolve, and hopefully also be grateful that they do. I am not raising this question because I want JSON schema to stop changing.

I'm asking this question because the messaging of the JSON schema project is very heavy on it being a draft. If the project considers the draft released as actual releases, then it would be good make it clear that it is not a draft in any sense other than some IETF standardization process, which may or may not complete, but should ultimately not be relevant because the right source of truth is not an IETF standard, as it already has already not been the source of truth through through the quite widespread adoption of various versions drafts/versions of the standard.

I also think it would be better to use semantic versioning instead of the date based versioning scheme, as this conveys more information to users of the standard. Even if every release bumped the major version (i.e. 2019-09 -> 8.0, 2020-12 -> 9.0) it would already be preferable than the current approach.

I do realize however that @Relequestual will address some of these issues in his talk this week and I will attend and raise any remaining questions I have afterwards.

**Reply by jdesrosiers** [reply]
I don't think standardization necessarily means that it will never change. For example, HTTP is on it's 5th version at this point. Although the project started off as an IETF draft with the goal of becoming an RFC, what we've effectively become is an independent group self publishing standards but still calling them drafts. We actually don't have any affiliation with IETF and never have.

I totally agree that the website is misleading. We're a small group of volunteers working in our free time. There is no one whose job it is to maintain the website and update the messaging. What usually happens is that someone like you comes along and says, "this doesn't make sense", then we say, "yep, you're right", and if someone happens to have the time and inclination to fix it, it might get updated. Honestly, often it doesn't because we all have other priorities at the time and it slips through the cracks.

**Reply by handrews** [reply]
@aucampia semantic versioning is not really suited to JSON Schema.  Draft releases and meta-schema releases do not go in lockstep with each other, and the meta-schema itself provides machine-readable compatibility information far more precisely than semantic versioning, which would just always be a major version which is worse than useless in my opinion.  Dates at least convey useful information.  There's hardly anything useful for us in the minor category, and we don't release new meta-schemas for the patch category.
